airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/parquet/GCSParquetWriter.java

```diff
@@ -0,0 +1,154 @@
+
+public class GCSParquetWriter extends BaseGCSWriter implements GCSWriter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(GCSParquetWriter.class);
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+  private static final ObjectWriter WRITER = MAPPER.writer();
+
+  private final Schema schema;
+  private final JsonFieldNameUpdater nameUpdater;
+  private final ParquetWriter<Record> parquetWriter;
+  private final JsonAvroConverter converter = new JsonAvroConverter();
+
+  public GCSParquetWriter(GCSDestinationConfig config,
+                         AmazonS3 s3Client,
+                         ConfiguredAirbyteStream configuredStream,
+                         Timestamp uploadTimestamp,
+                         Schema schema,
+                         JsonFieldNameUpdater nameUpdater)
+      throws URISyntaxException, IOException {
+    super(config, s3Client, configuredStream);
+    this.schema = schema;
+    this.nameUpdater = nameUpdater;
+
+    String outputFilename = BaseGCSWriter.getOutputFilename(uploadTimestamp, GCSFormat.PARQUET);
+    String objectKey = String.join("/", outputPrefix, outputFilename);
+
+    LOGGER.info("Full GCS path for stream '{}': {}/{}", stream.getName(), config.getBucketName(),
+        objectKey);
+
+    URI uri = new URI(
+        String.format("s3a://%s/%s/%s", config.getBucketName(), outputPrefix, outputFilename));    // <----- CHECK
+    Path path = new Path(uri);
+
+    GCSParquetFormatConfig formatConfig = (GCSParquetFormatConfig) config.getFormatConfig();
+    Configuration hadoopConfig = getHadoopConfig(config);
+    this.parquetWriter = AvroParquetWriter.<GenericData.Record>builder(HadoopOutputFile.fromPath(path, hadoopConfig))
+        .withSchema(schema)
+        .withCompressionCodec(formatConfig.getCompressionCodec())
+        .withRowGroupSize(formatConfig.getBlockSize())
+        .withMaxPaddingSize(formatConfig.getMaxPaddingSize())
+        .withPageSize(formatConfig.getPageSize())
+        .withDictionaryPageSize(formatConfig.getDictionaryPageSize())
+        .withDictionaryEncoding(formatConfig.isDictionaryEncoding())
+        .build();
+  }
+
+  public static Configuration getHadoopConfig(GCSDestinationConfig config) {
+    Configuration hadoopConfig = new Configuration();
+
+    // hadoopConfig.set("fs.gs.project.id", config.getProjectID());
+    // hadoopConfig.set("google.cloud.auth.service.account.email", "simphony-elt@data-247212.iam.gserviceaccount.com");
+    // hadoopConfig.set("fs.gs.auth.service.account.private.key.id", config.getAccessKeyId());
+    // hadoopConfig.set("fs.gs.auth.service.account.private.key", config.getSecretAccessKey());
+    // hadoopConfig.set("fs.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem"); 
+    // hadoopConfig.set("fs.AbstractFileSystem.gs.impl", "com.google.cloud.hadoop.fs.gcs.GoogleHadoopFS");
+    // hadoopConfig.set("google.cloud.auth.service.account.enable", "true");
```

Comment:
All the configs associated with GCS are commented out. Currently this writer only works for S3. Maybe you forgot to push the latest changes?

Reply:
hey @tuliren my bad, these are different tests I was doing to check how Airbyte was connecting to GCS: all those are useful. in case of service account (username, password and authkey).
I can test it again just to make sure I didn't mess up with the files while creating the PR, but it does work with GCS

After my test, i'll remove what is not needed.